assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js

```diff
@@ -114,7 +114,51 @@ describe('Upgrade Flow - Backup Controller', function() {
             });
 
             describe('when executed', function () {
+                describe('on success with wrong file headers', function () {
+                    var downloadBackupMock;
+
+                    beforeEach(function () {
+
+                        // Mock the click() method of a fake downloadBackup element
+                        downloadBackupMock = $document[0].createElement('a')
+                        spyOn(downloadBackupMock, 'click');
+
+                        // Mock the createElement() method of $document[0]
+                        spyOn($document[0], 'createElement')
+                            .and.returnValue(downloadBackupMock);
+
+                        // Mock the headers() method of the fake response
+                        spyOn(mockedDownloadResponse, 'headers')
+                            .and.returnValue({});
+
+                        // Mock the download() method of the crowbarBackupFactory,
+                        // and return a custom promise instead
+                        bard.mockService(crowbarBackupFactory, {
+                            get: $q.when(mockedDownloadResponse)
+                        });
+
+                        // Run the backup get function
+                        controller.backup.download(42);
+                        $rootScope.$digest();
+                    });
 
+                    it('click() method should have been triggered to download the backup', function () {
+                        expect($document[0].createElement).toHaveBeenCalledWith('a');
+                        expect(downloadBackupMock.click).toHaveBeenCalled();
```

Comment:
Could you also add `expect(downloadBackupMock.click.calls.count()).toBe(1);` so we assure only once this is being called?